nvflare/private/fed/app/simulator/simulator_worker.py

```diff
@@ -79,13 +82,24 @@ def do_one_task(self, client):
                 client_runner = fl_ctx.get_prop(FLContextKey.RUNNER)
                 self.fire_event(EventType.SWAP_IN, fl_ctx)
 
-                interval, task_processed = client_runner.fetch_and_run_one_task(fl_ctx)
-                self.logger.info(f"Finished one task run for client: {client.client_name}")
-
-                # if any client got the END_RUN event, stop the simulator run.
-                if client_runner.end_run_fired or client_runner.asked_to_stop:
-                    stop_run = True
-                    self.logger.info("End the Simulator run.")
+                run_task_tries = 0
+                while True:
+                    interval, task_processed = client_runner.fetch_and_run_one_task(fl_ctx)
+                    self.logger.info(f"Finished one task run for client: {client.client_name}")
```

Comment:
Maybe should include "interval" and "task_processed" in the log message so we know whether it is retry or the task is really done.

Reply:
Added